mods/numbers/test/numbers.unit.test.ts

```diff
@@ -0,0 +1,177 @@
+            )}
+        }
+      });
+     let result = await action.getNumber('ref');
+      
+    expect(stubNumber.calledOnce).to.be.equal(true)
+    expect(numberReturn).to.be.equal(result);
+  })
+
+  it('Deleting a number', async () =>{
+    const refReturn = {} 
+    sandbox.spy();
+    const stubNumber = sandbox.stub(FonosService.prototype,'getService').returns({
+        deleteNumber:() =>{
+          return{
+            sendMessage: ((r: any) => Promise.resolve(refReturn)
+            )}
+        }
+      });
+     let result = await numbers.deleteNumber('ref');
+      
+    expect(stubNumber.calledOnce).to.be.equal(true)
+    expect(refReturn).to.be.equal(result);
+  })
+
+  it('Listing a number', async () =>{
+     const stubNumber = sandbox.stub(FonosService.prototype,'getService').returns({
+        listNumbers:() =>{
+          return{
+            sendMessage: ((r: any) => Promise.resolve([])
+            )}
+        }
+      });
+     let result = await numbers.listNumbers({pageSize:0,pageToken:0,view:0});      
+    expect(stubNumber.calledOnce).to.be.equal(true)
+  })
+
+  it('Should return error with aorLink and ingressApp', async () =>{
+    let request ={
+        aorLink:"test",
+        ingressApp:"test"
+    };
+    const stubDb = sandbox.stub(numbers,'getNumber').returns({})
+    const numberSpy = sinon.spy(numbers,'updateNumber')
+    try{
+        await numbers.updateNumber(request);
+    }catch(err){
+        expect(err.message).to.include('are not compatible parameters')
+
+    }
+ })
+ it('Should return error with no aorLink and ingressApp', async () =>{
+    let request ={
+    
+    };
+    const stubDb = sandbox.stub(numbers,'getNumber').returns({})
+    const numberSpy = sinon.spy(numbers,'updateNumber')
+    try{
+        await numbers.updateNumber(request);
+    }catch(err){
+        expect(err.message).to.include('You must provider either')
+
+    }
+ })
+ it('Should udpdate a number with aorLink', async () =>{
+    let request ={
+        aorLink:"x"
+    };
+    const returnNumberDb = new NumbersPB.Number();
+    const stubDb = sandbox.stub(numbers,'getNumber').returns(returnNumberDb)
+    const stubNumber = sandbox.stub(FonosService.prototype,'getService').returns({
+        updateNumber:() =>{
+          return{
+            sendMessage: ((r: any) => Promise.resolve(returnNumberDb)
+            )}
+        }
+      });
+    let result = await numbers.updateNumber(request);
+    expect(result).to.be.equal(returnNumberDb);
+
+ })
+ it('Should udpdate a number with ingressApp', async () =>{
+    let request ={
+        ingressApp:"test"
+    };
+    const returnNumberDb = new NumbersPB.Number();
+    const stubDb = sandbox.stub(numbers,'getNumber').returns(returnNumberDb)
+    const stubNumber = sandbox.stub(FonosService.prototype,'getService').returns({
+        updateNumber:() =>{
+          return{
+            sendMessage: ((r: any) => Promise.resolve(returnNumberDb)
```

Comment:
Since you are not using `r` to compute the result, you can simply remove it from the parameter.